src/core/src/op/scatter_elements_update.cpp

```diff
@@ -273,3 +280,73 @@ bool op::v3::ScatterElementsUpdate::evaluate_label(TensorLabelVector& output_lab
     return ov::default_label_evaluator(this, {0, 2}, output_labels);
     OPENVINO_SUPPRESS_DEPRECATED_END
 }
+
+namespace ov {
+op::v12::ScatterElementsUpdate::ScatterElementsUpdate(const Output<Node>& data,
+                                                      const Output<Node>& indices,
+                                                      const Output<Node>& updates,
+                                                      const Output<Node>& axis,
+                                                      const Reduction reduction,
+                                                      const bool use_init_val)
+    : Op({data, indices, updates, axis}),
+      m_reduction{reduction},
+      m_use_init_val{use_init_val} {
+    constructor_validate_and_infer_types();
+}
+
+bool op::v12::ScatterElementsUpdate::visit_attributes(AttributeVisitor& visitor) {
+    OV_OP_SCOPE(v12_ScatterElementsUpdate_visit_attributes);
+    visitor.on_attribute("reduction", m_reduction);
+    visitor.on_attribute("use_init_val", m_use_init_val);
+    return true;
+}
+
+void op::v12::ScatterElementsUpdate::validate_and_infer_types() {
+    OV_OP_SCOPE(v12_ScatterElementsUpdate_validate_and_infer_types);
+    validate_scatter_element_types(this);
+
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    const auto output_shape = shape_infer(this, get_node_input_partial_shapes(*this)).front();
+    OPENVINO_SUPPRESS_DEPRECATED_END
+    set_output_type(0, get_input_element_type(0), output_shape);
```

Comment:
Consider set the merge result instead input_0 type (maybe is possible to change it for v3 also). In case when data is dynamic and updates are e.g. i32 the merge result of element type will be i32 and will give more accurate output.

The dynamic type can be only set when both are dynamic.